src/lib/Toast/Toast.svelte

```diff
@@ -0,0 +1,149 @@
+<script lang="ts">
+    import { fly } from 'svelte/transition';
+  
+    import { defaultToastProperties, type ToastProperties } from './properties';
+    import errorIcon from './error.svg';
+    import closeIcon from './close.svg';
+    import infoIcon from './info.svg'
+    import { onMount } from 'svelte';
+  
+    export let properties: ToastProperties = defaultToastProperties;
+  
+    let visible = false;
+  
+    $: leftIcon = properties.leftIcon ? properties.leftIcon : '';
+  
+    properties = {
+      ...defaultToastProperties,
+      ...properties
+    };
+  
+    onMount(() => {
+      visible = true
+      setTimeout(() => {
+        visible = false;
+      }, properties.duration);
+    });
+  
+    function getTypeClass(type: string): string {
+      switch (type) {
+        case 'success':
+          if (leftIcon.length == 0) {
+            leftIcon = 'https://sdk.breeze.in/gallery/icons/order-success-icon.svg';
+          }
+          return 'success-toast';
+        case 'error':
+          if (leftIcon.length == 0) {
+            leftIcon = errorIcon;
+          }
+          return 'error-toast';
+        case 'info':
+          if (leftIcon.length == 0) {
+            leftIcon = infoIcon;
+          }
+          return 'info-toast';
+        case 'warn':
+          if (leftIcon.length == 0) {
+            leftIcon = 'https://sdk.breeze.in/gallery/icons/warning.svg';
+          }
+          return 'warn-toast';
+        default:
+          return '';
+      }
+    }
+  </script>
+  
+  {#if visible}
+    <div
+      class="toast {getTypeClass(properties.type)}"
+      style="border-radius: {properties.borderRadius};"
+      in:fly={{ x: 500, y: 0, duration: 400 }}
+      out:fly={{ x: 500, y: 0, duration: 800 }}
+    >
+      <div class="toast-icon-container">
+        <div class="toast-icon-wrapper">
+          <img class="toast-icon" src={leftIcon} alt="toast-icon" />
+        </div>
+        <div style="color: {properties.textColor}">
+          {properties.message}
+        </div>
+      </div>
+  
+      {#if properties.showRightIcon}
+        <div>
+          <!-- svelte-ignore a11y-click-events-have-key-events -->
+          <!-- svelte-ignore a11y-no-static-element-interactions -->
```

Comment:
with proper on:keydown one warning can be removed

Reply:
updated.